internal/database/database.go

```diff
@@ -22,7 +22,12 @@ import (
 	"github.com/google/uuid"
 )
 
+type DatabaseReadWriter interface {
```

Comment:
The only function that needs `Delete` is the `stop` function which only needs a struct implementing Delete.
Why not just do a `DatabaseDelete` interface (or `Deleter`) and then do a simple type-case to that interface in the only case we need it?

```
var err error
client, ok := h.database.Open(r.Context(), identifier).(Deleter)
if !ok {
    logger.Warning("the database does not support delete, writing nil")
    err = client.Write(nil)  // Write a nil value to the database, don't delete.
} else {
    err = client.Delete()
}
if err != nil {
    // blabla
}
```

This way our databases don't need to support Delete all the time.

Reply:
If one implementation does not support deleting, the Delete() method can be left empty. But generally databases will have some kind of deleting.